src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+    public final String value;
+    public boolean isPrivate;
+    
+    public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        this.value = contact;
+    }
+    
+    protected Contact(){  
```

Comment:
What is the point of this constructor? Would it be better to call the other constructor from the sub classes? 
This is unnecessary.